lib/modules/manager/mix/artifacts.ts

```diff
@@ -26,15 +29,51 @@ export async function updateArtifacts({
   config,
 }: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
   logger.debug(`mix.getArtifacts(${packageFileName})`);
-  if (updatedDeps.length < 1) {
-    logger.debug('No updated mix deps - returning null');
+  const isLockFileMaintenance = config.updateType === 'lockFileMaintenance';
+
+  if (is.emptyArray(updatedDeps) && !isLockFileMaintenance) {
+    logger.debug('No updated mix deps');
+    return null;
+  }
+
+  let lockFileName = getSiblingFileName(packageFileName, 'mix.lock');
+  let isUmbrella = false;
+
+  let existingLockFileContent = await readLocalFile(lockFileName, 'utf8');
+  if (!existingLockFileContent) {
+    const lockFileError = await lockFileReadError(lockFileName);
+    if (lockFileError) {
+      return lockFileError;
+    }
+
+    const parentLockFileName =
+      (await findLocalSiblingOrParent(packageFileName, 'mix.lock')) ??
+      'mix.lock';
+    existingLockFileContent = await readLocalFile(parentLockFileName, 'utf8');
+
+    if (existingLockFileContent) {
+      lockFileName = parentLockFileName;
+      isUmbrella = true;
+    } else {
+      const lockFileError = await lockFileReadError(parentLockFileName);
+      if (lockFileError) {
+        return lockFileError;
+      }
+    }
+  }
+
+  if (isLockFileMaintenance && isUmbrella) {
+    logger.debug(
+      'Cannot use lockFileMaintenance in an umbrella project, see https://docs.renovatebot.com/modules/manager/mix/#lockFileMaintenance',
+    );
```

Comment:
@rarkins should we use a different log level here? i'm unsure.

Reply:
I don't mind too much about debug logs